persist/fs/msgpack/roundtrip_test.go

```diff
@@ -169,6 +170,53 @@ func TestIndexEntryRoundtrip(t *testing.T) {
 	require.Equal(t, testIndexEntry, res)
 }
 
+// Make sure the new decoding code can handle the old file format
+func TestIndexEntryRoundTripBackwardsCompatibilityV1(t *testing.T) {
+	var (
+		enc = newEncoder(newEncoderOptions{encodeLegacyV1IndexEntry: true})
+		dec = testDecoder(t, nil)
+	)
+
+	// Set the default values on the fields that did not exist in V1
+	// and then restore them at the end of the test - This is required
+	// because the new decoder won't try and read the new fields from
+	// the old file format
+	oldEncodedTags := testIndexEntry.EncodedTags
+	testIndexEntry.EncodedTags = nil
+	defer func() {
+		testIndexEntry.EncodedTags = oldEncodedTags
+	}()
+
+	enc.EncodeIndexEntry(testIndexEntry)
+	dec.Reset(NewDecoderStream(enc.Bytes()))
+	res, err := dec.DecodeIndexEntry()
+	require.NoError(t, err)
+	require.Equal(t, testIndexEntry, res)
+}
+
+// Make sure the old decoder code can handle the new file format
+func TestIndexEntryRoundTripForwardsCompatibilityV2(t *testing.T) {
+	var (
+		enc = newEncoder(newEncoderOptions{encodeLegacyV1IndexEntry: false})
+		dec = testDecoder(t, nil)
+	)
+
+	// Set the default values on the fields that did not exist in V1
+	// and then restore them at the end of the test - This is required
+	// because the old decoder won't read the new fields
+	oldEncodedTags := testIndexEntry.EncodedTags
```

Comment:
I think I did the same thing in my test, but I was dealing with integer types that would get encoded anyways. Do you want to encode data with tag data included, and then strip them from the test object for the sake of comparison? That way we know the decoder can actually skip over tags data if it is present

Reply:
Sure thing.